packages/integration/test/load/txserviceConcurrency.ts

```diff
@@ -0,0 +1,125 @@
+import pino from "pino";
+import PriorityQueue from "p-queue";
+import { ChainConfig, TransactionService, WriteTransaction } from "@connext/nxtp-txservice";
+import { RequestContext } from "@connext/nxtp-utils";
+import { BigNumber } from "ethers";
+
+import { TransactionInfo } from "../utils/sdkManager";
+import { getConfig } from "../utils/config";
+import { OnchainAccountManager } from "../utils/accountManager";
+
+/**
+ * Sets up a basic concurrency test through the core TransactionService only. Will slowly add more agents up to `maxConcurrency` sending transactions.
+ *
+ * @param maxConcurrency - Concurrency to build up to. We start at 1, and work up to this number. On each iteration, the concurrency value will be the number of agents who will do transactions simultaneously.
+ * @param step - Step by which we increase the concurrency.
+ */
+const txserviceConcurrencyTest = async (maxConcurrency: number, step = 1): Promise<void> => {
+  const config = getConfig();
+  const logger = pino({ level: config.logLevel ?? "info" });
+
+  // Just pick the first chain in the config.
+  const chainId = parseInt(Object.keys(config.chainConfig)[0]);
+  // Grab the first asset ID for this chain we can find.
+  // TODO: Will be used to get on chain balance after transfer.
+  // const assetId: string = ((): string => {
+  //   for (const swap of config.swapPools) {
+  //     for (const asset of swap.assets) {
+  //       if (asset.chainId === chainId) {
+  //         return asset.assetId;
+  //       }
+  //     }
+  //   }
+  //   throw new Error(`Could not find assetId for chainId: ${chainId}`);
+  // })();
+
+  // Create manager
+  const manager = new OnchainAccountManager(config.chainConfig, config.mnemonic, maxConcurrency);
+  logger.info({ agents: maxConcurrency }, "Created manager");
+
+
+  const chains: { [chainId: string]: ChainConfig } = {};
+  Object.entries(config.chainConfig).forEach(([chainId, config]) => {
+    chains[chainId] = {
+      confirmations: config.confirmations,
+      providers: config.providerUrls.map((url) => ({ url })),
+    } as ChainConfig;
+  });
+
+  // Set up a transaction service instance.
+  const txservice = new TransactionService(
+    logger,
+    manager.funder,
+    { chains },
+  );
+
+  logger.info("Beginning concurrency test.");
+  let concurrency: number;
+  let loopStats;
+  for (concurrency = 1; concurrency <= maxConcurrency; concurrency += step) {
+    // Create a queue to hold all payments with the given
+    // concurrency
+    const queue = new PriorityQueue ({ concurrency });
+
+    const tasks = Array(concurrency)
+      .fill(0)
+      .map((_, i) => {
+        const task = async () => {
+          const agent = manager.getRandomWallet();
+          const amount = "100000";
+          const txInfo: TransactionInfo = {
+            start: Date.now(),
+          };
+          try {
+            await txservice.sendTx(
```

Comment:
Do you think it would make sense to test for different types of states that can get returned implicitly, like should we make sure this hits a contract in a way that we know returns a call_revert for example. Or is that outside the wheelhouse of txservice's concerns?

Reply:
That's not really in scope here bc we're just sending basic tx's, not contract calls (def no transaction manager contract calls), really

Reply:
That *should* eventually be a part of integration testing though